lab_1_keywords_tfidf/main.py

```diff
@@ -155,11 +116,16 @@ def calculate_tf(frequencies: dict[str, int]) -> Optional[dict[str, float]]:
 
     In case of corrupt input arguments, None is returned
     """
-    if not check_dict(frequencies, str, int, False):
+
+    if not isinstance(frequencies, dict) or not frequencies:
         return None
-    sum_freq = sum(frequencies.values())
-    tf_dict = {word: (frequency / sum_freq) for word, frequency in frequencies.items()}
-    return tf_dict
+    for word in frequencies.keys():
+        if not isinstance(word, str):
+            return None
+    dict_tf = {}
+    for token, count in frequencies.items():
+        dict_tf[token] = count / sum(frequencies.values())
```

Comment:
it is better to compute sum of values once and for all, this way your code will be more efficient